docs/cloud-edge/modules/request-headers.mdx

```diff
@@ -0,0 +1,349 @@
+apiVersion: ingress.k8s.ngrok.com/v1alpha1
+metadata:
+  name: ngrok-module-set
+modules:
+  headers:
+    request:
+      add:
+        is-ngrok: "1"
+        country: "${.ngrok.geo.country_code}"
+      remove: ["referrer"]
+---
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: example-ingress
+  annotations:
+    k8s.ngrok.com/modules: ngrok-module-set
+spec:
+  ingressClassName: ngrok
+  rules:
+    - host: your-domain.ngrok.app
+      http:
+        paths:
+          - path: /
+            pathType: Prefix
+            backend:
+              service:
+                name: example-service
+                port:
+                  number: 80
+```
+
+### Edges
+
+Request Headers is a supported HTTP Edge Route module.
+
+- [Request Headers Edge Module API
+  Resource](/api/resources/https-edge-route-request-headers-module/)
+
+## Behavior
+
+### Ordering
+
+Request header changes made by other modules can be overridden by this module
+because this module is executed immediately before the HTTP request is
+transmitted to the upstream application.
+
+### Variable Interpolation
+
+You may interpolate variables into header values. Variables are interpolated
+into headers with JSONPath expressions surrounded by the `${}` syntax.
+
+For example to include geographical data about the client IP that initiated the
+request, you may construct a header value like so.
+
+```bash
+ngrok http 80 --request-header-add 'country: ${.ngrok.geo.country_code}'
+```
+
+If you are specifying variable interpolation from the command line, make sure
+to use single quotes for the command line argument otherwise it is likely that
+the shell will interpret your variable definition.
+
+Consult the [Variables Reference](#variables) for the available variables.
+
+### Multiple Header Values
+
+HTTP headers may include the same header multiple times. You may add a header
+multiple times with different values and it will be added multiple times. For
+example:
+
+```bash
+ngrok http --request-header-add "foo: bar" --request-header-add "foo: baz"
+```
+
+will result in a header with multiple values set
+
+```http
+GET / HTTP/1.1
+host: example.ngrok.app
+foo: bar
+foo: baz
+```
+
+:::note
+
+There is a bug which currently causes the above behavior to not be correct.
+Only the last header to be used when specifying multiple headers. This behavior
+will be fixed to match what is documented above.
+
+:::
+
+If you remove a header that has multiple values, all values will be removed.
+
+### Replacing Header Values
+
+If you add a header that is already present in the HTTP request, it will add
+another header. For example, if you run:
+
+```bash
+ngrok http --request-header-add "foo: new-value"
+```
+
+And the HTTP request was already:
+
+```http
+GET / HTTP/1.1
+host: example.ngrok.app
+foo: original-value
+```
+
+The client will receive the following:
+
+```http
+GET / HTTP/1.1
+host: example.ngrok.app
+foo: original-value
+foo: new-value
+```
```

Comment:
this example, this would not contain both values. it would only contain the `foo: new-value` header due to the bug above.